text/0000-add-scratch-stack.md

```diff
@@ -0,0 +1,161 @@
+
+## Motivation
+
+Some applications consist of a single
+[statically-linked](https://en.wikipedia.org/wiki/Static_library) executable.
+This is most common in golang, but is also often utilized in rust. It is also
+common in languages/toolchains not supported by Paketo like C/C++. These
+statically-linked binaries can be run on a minimally-modified empty container
+image (AKA "scratch" image) with few-to-no packages/libraries. Scratch images
+are a common choice of container image for statically-linked executables as it
+minimizes the size (in MB) of the image as well as the number of installed
+packages - in turn reducing the surface area for security issues.
+
+Currently, in order to build these statically-linked applications using Paketo
+buildpacks (and aiming for the smallest possible resultant run image) users
+must either opt for the [tiny
+stack](https://github.com/paketo-buildpacks/jammy-tiny-stack) or build their
+own scratch stack. We have evidence of users doing both of these things.
+Neither are satisfactory to users, however, as the tiny stack contains more
+packages than is necessary for those applications, and users would prefer not
+to maintain their own stacks if Paketo is willing and able to do so.
+
+Perhaps most importantly, we are [considering adding more packages to the tiny
+stack](https://github.com/paketo-buildpacks/rfcs/pull/231/files), and the
+biggest objection to that proposal is that it will further increase the size
+and surface area of the tiny image for statically-linked binaries. This could,
+in-turn, make Paketo less attractive for developers of these applications.
+
+Creating a scratch stack will decouple these use cases, allowing us to tailor
+the tiny stack to applications which want more packages, while also providing a
+better experience for statically-linked binaries.
+
+## Detailed Explanation
+
+### Repository/Image Location
+
+We will create a new Github repository at
+`paketo-buildpacks/jammy-scratch-stack` and a new pair of images on Dockerhub
+at `paketobuildpacks/build-jammy-scratch` and
+`paketobuildpacks/run-jammy-scratch`.
+
+We will adopt the tag system used for Jammy - specifically semver (i.e.
+'X.Y.Z') and 'latest' tags.
+
+We will not create a builder repository or image, though we reserve the option
+for the Builder team to do so in the future if that is requested and approved.
+
+### Stack ID
+
+We propose creating a new stack ID: `io.buildpacks.stacks.jammy.scratch`.
+This may require approval from the upstream Cloud-Native Buildpacks project.
+
+### Image Contents
+
+The scratch stack run image will contain the `tzdata` and `ca-certificates`
+packages, the standard CNB/Paketo metadata (including users and /etc/passwd
+entries), and nothing else. These two packages are often provided in scratch
+images as they are small and have minimal security risk.
+
+The build image will contain the same packages as the tiny stack (probably with
```

Comment:
I don't think it will save much work because pragmatically we will leverage `jam create-stack` which I believe requires a `Dockerfile` for both `build` and `run`, and will build both images every time.

We could explore a way to instruct `jam create-stack` to instead use an upstream image, but I think that's out of the scope of this RFC.